pkg/common/utils/resource/pod.go

```diff
@@ -518,3 +525,49 @@ func getProbe(port int32, path string) corev1.ProbeHandler {
 
 	return p
 }
+
+func getDefaultAffinity(componentType v1.ComponentType) *corev1.Affinity {
+	// default Affinity rule is :
+	// Pods of the same component should deploy on different hosts with Preferred scheduling.
+	// weight is 20, weight range is 1-100
+	podAffinityTerm := corev1.WeightedPodAffinityTerm{
+		Weight: 20,
+		PodAffinityTerm: corev1.PodAffinityTerm{
+			LabelSelector: &metav1.LabelSelector{
+				MatchExpressions: []metav1.LabelSelectorRequirement{
+					{Key: v1.ComponentLabelKey, Operator: metav1.LabelSelectorOpIn, Values: []string{string(componentType)}},
+				},
+			},
+			TopologyKey: "kubernetes.io/hostname",
+		},
+	}
+
+	podAntiAffinity := corev1.PodAntiAffinity{
+		PreferredDuringSchedulingIgnoredDuringExecution: []corev1.WeightedPodAffinityTerm{podAffinityTerm},
+	}
+
+	return &corev1.Affinity{
+		PodAntiAffinity: &podAntiAffinity,
+	}
+}
+
+func GetAffinity(dcrAffinity *corev1.Affinity, componentType v1.ComponentType) *corev1.Affinity {
+	affinity := getDefaultAffinity(componentType)
+
+	if dcrAffinity == nil {
+		return affinity
+	}
+
+	affinity.NodeAffinity = dcrAffinity.NodeAffinity
+	affinity.PodAffinity = dcrAffinity.PodAffinity
+	dcrPodAntiAffinity := dcrAffinity.PodAntiAffinity
+
+	if dcrPodAntiAffinity != nil {
```

Comment:
This if not needed, above if dcrAffinity== nil have return, next logical deal dcrAffinity != nil.